src/WebJobs.Script/Http/RetryProxyHandler.cs

```diff
@@ -30,11 +32,22 @@ public RetryProxyHandler(HttpMessageHandler innerHandler, ILogger logger)
 
         protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
+            TaskCompletionSource<ScriptInvocationResult> resultSource = null;
+            if (request.Options.TryGetValue(ScriptConstants.HttpProxyScriptInvocationContext, out ScriptInvocationContext scriptInvocationContext))
+            {
+                resultSource = scriptInvocationContext.ResultSource;
+            }
+
             var currentDelay = InitialDelay;
             for (int attemptCount = 1; attemptCount <= MaxRetries; attemptCount++)
             {
                 try
                 {
+                    if (resultSource is not null && resultSource.Task.IsFaulted)
+                    {
+                        throw resultSource.Task.Exception?.InnerException ?? new InvalidOperationException("The function invocation tied to this HTTP request failed.");
```

Comment:
Are we saying we don't want to retry if the request failed? The second part of this statement will end up in the last exception catch (lines 72-80

Reply:
We don't want to retry if the invocation was failed on the functions, non-http side - without this change, the proxying service will keep retrying even though the host has stopped tracking the invocation. This will align the http service with the host by making sure to check that the invocation is still considered "valid" or "executing".